src/services/custom-zod-schemas.ts

```diff
@@ -49,7 +49,7 @@ export const optionalBooleanSchema = () =>
   z
     .string()
     .optional()
-    .transform((val) => val === "true");
+    .transform((val) => val === "on" || val === "true");
```

Comment:
todo: add a commentary to explain this.

- value "on" is send by form submission with checkbox ticked
- value "true" is send programatically on our side.

Another option could be to use a different custom zod schema `optionalCheckboxSchema`.